src/Diagnostics/FxCop/System.Runtime.Analyzers/CSharp/Globalization/UseOrdinalStringComparison.Fixer.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace System.Runtime.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class CSharpUseOrdinalStringComparisonFixer : UseOrdinalStringComparisonFixerBase
+    {
+        protected override bool IsInArgumentContext(SyntaxNode node)
+        {
+            return node.IsKind(SyntaxKind.Argument) &&
+                   ((ArgumentSyntax)node).Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression);
+        }
+
+        protected override Task<Document> FixArgument(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode argument)
+        {
+            var memberAccess = ((ArgumentSyntax)argument)?.Expression as MemberAccessExpressionSyntax;
```

Comment:
The ```?.``` seems superfluous here since ```FixArgument()``` will only be called if ```IsInArgumentContext()``` returns ```true```.

Could you change type for the ```node``` argument for ```FixArgument()``` to ```ArgumentSyntax``` in the base type so that we can avoid casting in the derived types?